application/src/main/java/org/togetherjava/tjbot/commands/moderation/WhoIsCommand.java

```diff
@@ -0,0 +1,263 @@
+            } else {
+                return handleWhoIsUser(event, user, profile);
+            }
+        }).queue();
+    }
+
+    @CheckReturnValue
+    private static @NotNull ReplyAction handleWhoIsUser(final @NotNull SlashCommandEvent event,
+            final @NotNull User user, final @NotNull User.Profile profile) {
+
+        StringBuilder descriptionBuilder =
+                new StringBuilder().append(handleUserIdentification(user))
+                    .append("\n**Is bot:** ")
+                    .append(user.isBot())
+                    .append(handleUserFlags(user.getFlags()))
+                    .append("\n**Registration date:** ")
+                    .append(DATE_TIME_FORMAT.format(user.getTimeCreated()));
+
+        EmbedBuilder embedBuilder =
+                generateEmbedBuilder(event, user, profile, profile.getAccentColor()).setAuthor(
+                        user.getName(), user.getEffectiveAvatarUrl(), user.getEffectiveAvatarUrl())
+                    .setDescription(descriptionBuilder);
+
+        return event.replyEmbeds(embedBuilder.build())
+            .addActionRow(Button.of(ButtonStyle.LINK, "discord://-/users/" + user.getId(),
+                    "Click to see profile"));
+    }
+
+    @CheckReturnValue
+    private static @NotNull ReplyAction handleWhoIsMember(final @NotNull SlashCommandEvent event,
+            final @NotNull Member member, final @NotNull User.Profile profile) {
+        User user = member.getUser();
+
+        Color memberColor = member.getColor();
+        Color effectiveColor = (null == memberColor) ? profile.getAccentColor() : memberColor;
+
+        StringBuilder descriptionBuilder =
+                new StringBuilder().append(handleUserIdentification(user))
+                    .append(handleVoiceState(member))
+                    .append("\n**Is bot:** ")
+                    .append(user.isBot())
+                    .append(handlePossibleBooster(member))
+                    .append(handleUserFlags(user.getFlags()))
+                    .append("\n**Join date:** ")
+                    .append(DATE_TIME_FORMAT.format(member.getTimeJoined()))
+                    .append("\n**Registration date:** ")
+                    .append(DATE_TIME_FORMAT.format(user.getTimeCreated()))
+                    .append("\n**Roles:** ")
+                    .append(formatRoles(member));
+
+
+        EmbedBuilder embedBuilder = generateEmbedBuilder(event, user, profile, effectiveColor)
+            .setAuthor(member.getEffectiveName(), member.getEffectiveAvatarUrl(),
+                    member.getEffectiveAvatarUrl())
+            .setDescription(descriptionBuilder);
+
+
+        return event.replyEmbeds(embedBuilder.build())
+            .addActionRow(Button.of(ButtonStyle.LINK, "discord://-/users/" + user.getId(),
+                    "Click to see profile"));
+    }
+
+    private static @NotNull String handleVoiceState(@NotNull final Member member) {
```

Comment:
The name seems a bit confusing to me. It seems to be more of a "voice state to string" method. Maybe you can find a better name.